vpr/src/route/clock_network_builders.cpp

```diff
@@ -640,7 +648,14 @@ int ClockSpine::create_chany_wire(int y_start,
     }
     node.set_cost_index(CHANX_COST_INDEX_START + num_segments + seg_index);
 
-    add_to_rr_node_indices(*rr_node_indices, *rr_nodes, node_index);
+    /* Add the node to spatial lookup */
```

Comment:
This code has been repeated almost exactly (minus the weird CHANX/CHANY x/y index swapping) 3 times.  Should be a helper function.  Maybe not so easy to make a helper due to the swapping of x/y, but should have a TODO to make a helper in that case.  Or (better) make the .add_node function do the x/y swapping based on the node type (CHANX swaps I think) so all 3 function calls look the same and you can make a helper function. Plus the weird swapping is then localized in the node_lookup class.

Reply:
Yes. I think the new API API ``add_node_to_all_locs()`` as you suggested in another PR #1800 will resolve this issue. 
I leave a TODO comment here to remind me to finish the API replacement after PR #1800 is merged.
Let me know if this is fine for you.

Reply:
Sounds good; thanks.